lib/obj2gltf.js

```diff
@@ -26,19 +24,23 @@ module.exports = obj2gltf;
  * @param {Boolean} [options.binary=false] Save as binary glTF.
  * @param {Boolean} [options.separate=false] Writes out separate geometry data files, shader files, and textures instead of embedding them in the glTF.
  * @param {Boolean} [options.separateTextures=false] Write out separate textures only.
- * @param {Boolean} [options.compress=false] Quantize positions, compress texture coordinates, and oct-encode normals.
- * @param {Boolean} [options.optimize=false] Optimize the glTF for size and runtime performance.
- * @param {Boolean} [options.optimizeForCesium=false] Optimize the glTF for Cesium by using the sun as a default light source.
- * @param {Boolean} [options.generateNormals=false] Generate normals if they are missing.
- * @param {Boolean} [options.ao=false] Apply ambient occlusion to the converted model.
- * @param {Boolean} [options.kmc=false] Output glTF with the KHR_materials_common extension.
- * @param {Boolean} [options.textureCompressionOptions] Options sent to the compressTextures stage of gltf-pipeline.
- * @param {Boolean} [options.bypassPipeline=false] Bypass the gltf-pipeline for debugging purposes. This option overrides many of the options above and will save the glTF with the KHR_materials_common extension.
  * @param {Boolean} [options.checkTransparency=false] Do a more exhaustive check for texture transparency by looking at the alpha channel of each pixel.
  * @param {Boolean} [options.secure=false] Prevent the converter from reading image or mtl files outside of the input obj directory.
  * @param {String} [options.inputUpAxis='Y'] Up axis of the obj. Choices are 'X', 'Y', and 'Z'.
  * @param {String} [options.outputUpAxis='Y'] Up axis of the converted glTF. Choices are 'X', 'Y', and 'Z'.
+ * @param {Boolean} [options.packOcclusion=false] Pack the occlusion texture in the red channel of metallic-roughness texture.
+ * @param {Boolean} [options.metallicRoughness=false] The values in the mtl file are already metallic-roughness PBR values and no conversion step should be applied. Metallic is stored in the Ks and map_Ks slots and roughness is stored in the Ns and map_Ns slots.
+ * @param {Boolean} [options.specularGlossiness=false] The values in the mtl file are already specular-glossiness PBR values and no conversion step should be applied. Specular is stored in the Ks and map_Ks slots and glossiness is stored in the Ns and map_Ns slots. The glTF will be saved with the KHR_materials_pbrSpecularGlossiness extension.
+ * @param {Boolean} [options.materialsCommon=false] The glTF will be saved with the KHR_materials_common extension.
+ * @param {String} [options.metallicRoughnessOcclusionTexture] Path to the metallic-roughness-occlusion texture used by the model, where occlusion is stored in the red channel, roughness is stored in the green channel, and metallic is stored in the blue channel. This may be used instead of setting texture paths in the .mtl file. The model will be saved with a pbrMetallicRoughness material.
```

Comment:
Do the `path to XXX texture` parameters disallow materials referencing multiple images in the obj? If so, we should document that users of these "advanced features" need to make sure the input texture covers those channels across the entire input obj.